clojure.md

```diff
@@ -301,6 +301,54 @@ keymap ; => {:a 1, :b 2, :c 3}
   (conj [4 5 6] input 8 9 10)) ;=> or in the middle !
                                ; Result: [4 5 6 4 8 9 10]
 
+; Destructuring
+;;;;;;;;;;;;;;;
+
+(let [[x y & more] [1 2 3 4]
+      {:keys [a b] :or {b 99}} {:a 42}]
+  [x y more a b])
+; => [1 2 (3 4) 42 99]
+
+; Loops
+;;;;;;;;;;;;;;;
+
+(loop [n 10 acc 0]
+  (if (zero? n)
+    acc
+    (recur (dec n) (+ acc n))))
+; => 55
+
+(for [x (range 5) :when (even? x)] (* x x))
+; => (0 4 16)
+
+(doseq [x (range 3)]
+  (println "side effect" x))
```

Comment:
This might deserve an explanation

Reply:
@vendethiel thanks, done.

Reply:
That looks like an explanation from the docs, but I thing the more important part is that it does all side effects